### PR TITLE
feat(hubspot): create contact from Charles

### PR DESCRIPTION
**Pull request by Dymidrol**

### CODE REVIEW

**Comment by eljefedelrodeodeljefe** [comment]
Task linked: [TECH-2294 Create HubSpot contact manually](https://app.clickup.com/t/24327122/TECH-2294)

File: src/entities/crm/crm-user.ts [hunk 1]
```diff
@@ -69,3 +69,7 @@ export interface AssociateUsersPayload {
   readonly externalUserReferenceId: string
   readonly staffId: string
 }
+
+export interface PersonPayload {
+  person: string
```
Comment (dafen) [inline src/entities/crm/crm-user.ts:74]: nit: not really necessary, you could also just take a string parameter directly.

File: src/entities/crm/crm.ts [hunk 2]
```diff
@@ -337,6 +341,32 @@ export class CRM extends UniverseEntity<CRMPayload, CRMRawPayload> {
       throw this.handleError(new CRMAssociateUsersError(undefined, { error: err }))
     }
   }
+
+  public async createExternalUserFromPerson (payload: PersonPayload): Promise<number | undefined> {
```
Comment (dafen) [inline src/entities/crm/crm.ts:345]: nit: see above: why not just accept a string value, called person?

File: src/entities/crm/crm.ts [hunk 3]
```diff
@@ -337,6 +341,32 @@ export class CRM extends UniverseEntity<CRMPayload, CRMRawPayload> {
...
+      person: payload.person
+    }
+
+    try {
+      const opts = {
+        method: 'PUT',
+        url: `${this.universe.universeBase}/${this.endpoint}/${this.id}/create_external_user_from_person`,
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8',
+          'Content-Length': '0'
```
Comment (dafen) [inline src/entities/crm/crm.ts:358]: Just remove this line, as content length is not 0, you pass an id.

File: src/entities/crm/crm.ts (hunk 2 above)
Comment (dafen) [reply src/entities/crm/crm.ts:345]: The promise response should not be a number, but the created channel user payload

**Review by dafen — CHANGES_REQUESTED** [review]
Small nits, and the please adapt the promise return when the endpoint will return a channel user.

File: src/entities/crm/crm.ts (hunk 3 above)
Comment (Dymidrol) [reply src/entities/crm/crm.ts]: done

File: src/entities/crm/crm.ts (hunk 2 above)
Comment (Dymidrol) [reply src/entities/crm/crm.ts]: done

File: src/entities/crm/crm-user.ts (hunk 1 above)
Comment (Dymidrol) [reply src/entities/crm/crm-user.ts]: done

File: src/entities/crm/crm.ts [hunk 4]
```diff
@@ -361,6 +389,14 @@ export class CRMSetupRemoteError extends BaseError {
   }
 }
 
+export class CRMCreateExternalUserFromPersonError extends BaseError {
```
Comment (eljefedelrodeodeljefe) [inline src/entities/crm/crm.ts:392]: nit: can you use BaseErrorV2 with a changed signature here please. This enables forwarding API errors to the UI, which would be useful here.

Comment (Dymidrol) [reply src/entities/crm/crm.ts:392]: Should I then refactor all crm file with BaseErrorV2 usage?

Comment (Dymidrol) [reply src/entities/crm/crm.ts]: did it for every error class

**Review by eljefedelrodeodeljefe — APPROVED** [review]

**Pull request merged**